server/aesdsocket.c

```diff
@@ -242,188 +567,132 @@ int main(int argc, char *argv[])
+                    timer_started = true;                
+                }
+            }
 
             //accept socket connections
-            sockfd_client = accept(sockfd_server, (struct sockaddr *)&new_addr, &addr_size);
+            sockfd_client = accept(sockfd_server, (struct sockaddr *)&client_addr, &addr_size);
             if (sockfd_client < 0)
             {
                 LOG_ERROR("error accepting connection\n");
                 server_socket_state = STATE_EXIT;
                 break;
             }
 
-            inet_ntop(AF_INET, &client_addr, str, INET_ADDRSTRLEN);
-            syslog(LOG_INFO, "Accepted connection from %s", str);
-
-            nwrite = 0, nreceived = 0, nsent = 0, nread = 0;
-            heap_buffer_for_write = NULL;
-            heap_buffer_for_read = NULL;
-            filled_up_buffer_size = 0;
-            total_buffer_size_for_write_into_file = BUF_SIZE;
-
-            heap_buffer_for_write = (char *)malloc(sizeof(char) * BUF_SIZE);
-            if (heap_buffer_for_write == NULL)
-            {
-                LOG_ERROR("could not malloc");
-                server_socket_state = STATE_EXIT;
-                break;
-            }
-            server_socket_state = STATE_RECEIVE_FROM_CLIENT;
+            LOG_DBG("CREATING THREAD\n");
+            //malloc for thread params
+            socket_slist_data_t *socket_server_data_ptr = NULL;
+            socket_server_data_ptr = (socket_slist_data_t *)malloc(sizeof(socket_slist_data_t));
+            socket_server_data_ptr->socket_thread_params.threadIdx = i++;
+            socket_server_data_ptr->socket_thread_params.socket_thread_completion_flag = false;
+            socket_server_data_ptr->socket_thread_params.fd = file_descriptor;
+            socket_server_data_ptr->socket_thread_params.sockfd_client = sockfd_client;
+            
+            pthread_create(&socket_server_data_ptr->socket_thread,                           // pointer to thread descriptor
+                           (void *)0,                                                        // use default attributes
+                           socket_communication,                                             // thread function entry point
+                           (void *)&(socket_server_data_ptr->socket_thread_params) // parameters to pass in
+            );
+
+            SLIST_INSERT_HEAD(&socket_head, socket_server_data_ptr, entries);
+
+            //inet_ntop(AF_INET, &client_addr, socket_server_data_ptr->socket_thread_params.str, INET_ADDRSTRLEN);
+            socket_server_data_ptr->socket_thread_params.str = inet_ntoa(client_addr.sin_addr);
+            
+            syslog(LOG_INFO, "Accepted connection from %s\n", socket_server_data_ptr->socket_thread_params.str);
+            LOG_DBG("Accepted connection from %s\n", socket_server_data_ptr->socket_thread_params.str);
+
+            pthread_join(socket_server_data_ptr->socket_thread, NULL);
```

Comment:
@rajatchaple you can't `pthread_join` here or your application won't actually use threading.  You need to wait until your thread is complete before you join.